framework/db/mssql/QueryBuilder.php

```diff
@@ -168,23 +168,26 @@ public function alterColumn($table, $column, $type)
     /**
      * Builds a SQL statement for enabling or disabling integrity check.
      * @param bool $check whether to turn on or off the integrity check.
-     * @param string $schema the schema of the tables. Defaults to empty string, meaning the current or default schema.
-     * @param string $table the table name. Defaults to empty string, meaning that no table will be changed.
+     * @param string $schema the schema of the tables.
+     * @param string $table the table name.
      * @return string the SQL statement for checking integrity
-     * @throws InvalidParamException if the table does not exist or there is no sequence associated with the table.
      */
     public function checkIntegrity($check = true, $schema = '', $table = '')
     {
-        if ($schema !== '') {
-            $table = "{$schema}.{$table}";
-        }
+        $enable = $check ? 'CHECK' : 'NOCHECK';
+        $schema = $schema ? $schema : $this->db->getSchema()->defaultSchema;
         $table = $this->db->quoteTableName($table);
-        if ($this->db->getTableSchema($table) === null) {
-            throw new InvalidParamException("Table not found: $table");
+        $tableNames = $this->db->getTableSchema($table) ? [$table] : $this->db->getSchema()->getTableNames($schema);
+        $viewNames = $this->db->getSchema()->getViewNames($schema);
+        $tableNames = array_diff($tableNames, $viewNames);
+        $command = '';
+
+        foreach ($tableNames as $tableName) {
+            $tableName = '"' . $schema . '"."' . $tableName . '"';
```

Comment:
I would quote `$schema` and `$tableName` with `$this->db->quoteColumnName()` - while `$table` could be left as is because we usually only need escaping (a value) when we then inject the value into SQL.